### PR TITLE
feat: 🎸 Strategy To load a Reality File URL Directly

### PR DESCRIPTION
**Pull request by pgobriensap**
- Logic in sample app to Extract Reality File from the app bundle and place it in Documents Directory
- New Reality File Strategy can load Reality File from a URL
- This will enable the reality File to be sourced from URL and now does not need to be bundled from Xcode

TODO:
- If JSON proposal is accepted then create an init to load from a JSON File

### CODE REVIEW

File: Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+//
+//  Created by O'Brien, Patrick on 3/2/21.
+//
+
+import ARKit
+import Foundation
+import RealityKit
+import SwiftUI
+
+/// A loading strategy that uses the RealityComposer app. After creating the Reality Composer scene, tthe entities in the scene correlate to a real world location relative to the image or object anchor.
```
Comment (MarcoEidinger) [inline Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift:13]: spelling error: tthe

File: Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+///
+/// ## Usage
+/// ```
+/// let cardItems = [ExampleCardItem(id: 0, title_: "Hello"), ExampleCardItem(id: 1, title_: "World")]
+/// guard let anchorImage = UIImage(named: "qrImage") else { return }
+/// let realityFilePath = FileManager.default.getDocumentsDirectory().appendingPathComponent(FileManager.realityFiles).appendingPathComponent("ExampleRC.reality")
+/// let strategy = RealityFileStrategy(cardContents: cardItems, anchorImage: anchorImage, physicalWidth: 0.1, realityFilePath: realityFilePath, rcScene: "ExampleScene")
+/// arModel.load(loadingStrategy: strategy)
+/// ```
+public struct RealityFileStrategy<CardItem: CardItemModel>: AnnotationLoadingStrategy where CardItem.ID: LosslessStringConvertible {
```
Comment (MarcoEidinger) [inline Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift:32]: Looks like duplicate code as most of it is present in `RealityComposerStrategy`. Only main difference is the `init` and different use of `RCScannner.load...` function. Recommend to consolidate reusable code across strategies

**Review by MarcoEidinger — CHANGES_REQUESTED** [review]

File: Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift (hunk 2 above)
Comment (pgobriensap) [reply Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift:32]: That is definitely fair. By consolidate, do you mean this should all just live within the RealityComposerStrategy with perhaps more inits and methods? There will be another URL which uses a usdz file that will be almost the exact same as this one as well.

Comment (MarcoEidinger) [reply Sources/FioriARKit/ARCards/RealityKit/RealityFileStrategy.swift:32]: There can be different strategy implementation but internally we should share reusable code. I favor composition over inheritance

**Pull request closed without merging**